lib/rexml/xpath_parser.rb

```diff
@@ -78,7 +78,9 @@ def variables=( vars={} )
 
     def parse path, nodeset
       path_stack = @parser.parse( path )
-      match( path_stack, nodeset )
+      nodeset.first&.document&.send(:enable_cache) do
```

Comment:
If the first node doesn't have a document, this returns `nil`. Is this backward compatible?

Reply:
`XPath.match`, `XPath.first` and `XPath.each` can accept both node and array, so this is possible.
```ruby
REXML::XPath.match([] ,'*')
REXML::XPath.first([] ,'*')
REXML::XPath.each([] ,'*'){}
```

This usage is not documented, but used in test codes.
https://github.com/ruby/rexml/blob/5d2606a99957f1623e464d70be065a31ed0ea58b/test/xpath/test_base.rb#L415
https://github.com/ruby/rexml/blob/5d2606a99957f1623e464d70be065a31ed0ea58b/test/test_jaxen.rb#L86

Currently, size of the array passed to match/first/each in the test code is always 1.

Reply:
Oh sorry...
I added `nodeset.empty?` test case.

> Similarly, I would expect an error if the first node does not have a document object.

I added a test case where the first node does not have a document object.